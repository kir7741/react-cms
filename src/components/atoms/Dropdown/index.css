.dropdown-wrapper {
	margin-bottom: 10px;
	padding-bottom: 1.25em;
	font-size: 16px;

	& > div {
		position: relative;
	}
}
