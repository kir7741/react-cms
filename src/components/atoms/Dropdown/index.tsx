import React, { useState } from 'react';
import classnames from 'classnames';

import styles from './index.module.css';

interface valueComponentProperty {

	/**
	 * 是否展開
	 *
	 * @type {boolean}
	 * @memberof valueComponentProperty
	 */
	isOpen: boolean;

}

interface DropdownProperty {
	className?: string;
	valueComponent: (props: valueComponentProperty) => JSX.Element,
	panelComponent: () => JSX.Element,
}

const Dropdown: React.FC<DropdownProperty> = ({
	className,
	valueComponent,
	panelComponent,
}) => {

	const [isOpen, setIsOpen] = useState(false);

	return (
		<div className={classnames(styles.dropdownWrapper, className)}>
			<div
				role="button"
				tabIndex={0}
				onKeyPress={() => {}}
				onClick={() => {
					setIsOpen(pre => !pre)
				}}
			>
				{valueComponent({ isOpen })}
				{
					isOpen && panelComponent()
				}
			</div>
		</div>
	);
};

export default Dropdown;
