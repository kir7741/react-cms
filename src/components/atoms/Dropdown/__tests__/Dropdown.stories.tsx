import React from 'react';
import { storiesOf } from '@storybook/react';

import Dropdown from 'components/atoms/Dropdown';

const stories = storiesOf('atoms/Dropdown', module);

stories.add('__interactive', () => <Dropdown valueComponent={() => <></>} panelComponent={() => <></>}/>);
