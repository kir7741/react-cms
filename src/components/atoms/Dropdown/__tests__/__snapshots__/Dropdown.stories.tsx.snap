// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="dropdownWrapper"
  >
    <div
      role="button"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;
